model_compression_toolkit/core/common/quantization/bit_width_config.py

```
@@ -32,6 +33,64 @@ class ManualBitWidthSelection:
+                        f"must match the number of filters {len(filters)}, or a single bit_width value "
+                        f"should be provided for all filters.")
+    elif len(vals) == 1 and len(filters) > 1:
+        vals = [vals[0] for f in filters]
+    return vals
+
+def _expand_to_list_filter_and_bit_width(
+        filters: Union[List[BaseNodeMatcher]],
+        bit_widths: Union[List[int], int],
+        attrs: Union[List[str], str] = None):
+    filters = [filters] if not isinstance(filters, list) else filters
+    bit_widths = _expand_to_list_core(filters, bit_widths)
+    attrs = _expand_to_list_core(filters, attrs)
+
+    return attrs, bit_widths, filters
+
+def _make_nodes_to_change_bit_width(graph, manual_bit_width_selection_list):
+    unit_nodes_to_change_bit_width = {}
+    for manual_bit_width_selection in manual_bit_width_selection_list:
+        filtered_nodes = graph.filter(manual_bit_width_selection.filter)
+        if len(filtered_nodes) == 0:
+            Logger.critical(
+                f"Node Filtering Error: No nodes found in the graph for filter {manual_bit_width_selection.filter.__dict__} "
+                f"to change their bit width to {manual_bit_width_selection.bit_width}.")
+        for n in filtered_nodes:
+            # check if a manual configuration exists for this node
+            if n in unit_nodes_to_change_bit_width:
+                Logger.info(
+                    f"Node {n} has an existing manual bit width configuration of {unit_nodes_to_change_bit_width.get(n)}. A new manual configuration request of {manual_bit_width_selection.bit_width} has been received, and the previous value is being overridden.")
+            if isinstance(manual_bit_width_selection_list,  ManualBitWidthSelection):
+                unit_nodes_to_change_bit_width.update({n: manual_bit_width_selection.bit_width})
+            elif isinstance(manual_bit_width_selection_list, ManualWeightsBitWidthSelection):
```

Something very important that is missing in this case - you must verify that the requested attribute to change the bit width for is existing in the node that was filtered. If not, the request is invalid, and you should throw an error.
You can do it by checking the attribute names returned from `node.get_node_weights_attributes()`

---

I added a process to verify whether the layer contains attributes using node.get_node_weights_attributes().